test/models/questionnaire_test.rb

```diff
@@ -402,12 +402,32 @@ class QuestionnaireTest < ActiveSupport::TestCase
       assert_nil questionnaire.checked_in_by_id
     end
 
-    should "return user who checked in ther questionnaire" do
+    should "return user who checked in their questionnaire" do
       user = create(:user)
       questionnaire = create(:questionnaire, checked_in_by_id: user.id)
       assert_equal user.id, questionnaire.checked_in_by.id
       assert_equal user.id, questionnaire.checked_in_by_id
     end
+
+    should "return deleted user who checked in their questionnaire" do
+      user = create(:user)
+      questionnaire = create(:questionnaire, checked_in_by_id: user.id)
+      old_user_id = user.id.dup
+      User.destroy(user.id)
```

Comment:
I'd also wrap this in a [`assert_difference`](https://api.rubyonrails.org/v5.2.4.2/classes/ActiveSupport/Testing/Assertions.html) to make sure this didn't somehow break the regular destroy logic:

```suggestion
      assert_difference 'User.count', 1 do
        User.destroy(user.id)
      end
```

Reply:
Though this should probably go in a `user_test.rb` specifically testing the destroy change